src/remarx/sentence/corpus/tei_input.py

```diff
@@ -95,6 +95,20 @@ def get_page_footnotes(self) -> list[xmlmap.XmlObject]:
 
         return page_footnotes
 
+    def get_line_number_for_position(
+        self, text: str, char_pos: int, is_footnote: bool = False
+    ) -> int:
+        """
+        Get the line number for a given character position in the text by counting newlines.
+        """
+
+        line_number = 1  # default to line 1
+        for i in range(min(char_pos, len(text))):
+            if text[i] == "\n":
+                line_number += 1
+
+        return line_number
+
```

Comment:
When iterating over text in `get_body_text()`:
- create a dictionary of line indices to track line number & character offset
- keep track of character offset as text is collected; move whitespace cleaning to inner loop so character offset is accurate for the cleaned up text

Revise this method to take character index; return the line break number for the line just before the character index for the start of the sentence.